src/wp-includes/block-template-utils.php

```diff
@@ -507,6 +507,64 @@ function _remove_theme_attribute_from_template_part_block( &$block ) {
 	}
 }
 
+/**
+ * Gets the contents for the given template file path, using a transient cache.
+ *
+ * @since 6.4.0
+ * @access private
+ *
+ * @param string $template_file_path Absolute path to a theme template file.
+ * @return string The template file contents.
+ */
+function _get_block_template_file_content( $template_file_path ) {
+	$theme = wp_get_theme();
+
+	$template_file_path = wp_normalize_path( $template_file_path );
+	$theme_dir          = wp_normalize_path( $theme->get_stylesheet_directory() ) . '/';
+
+	if ( str_starts_with( $template_file_path, $theme_dir ) ) {
+		$relative_path = substr( $template_file_path, strlen( $theme_dir ) );
+	} elseif ( $theme->parent() ) {
+		$theme     = $theme->parent();
+		$theme_dir = wp_normalize_path( $theme->get_stylesheet_directory() ) . '/';
+		if ( str_starts_with( $template_file_path, $theme_dir ) ) {
+			$relative_path = substr( $template_file_path, strlen( $theme_dir ) );
+		}
+	}
+
+	// Bypass cache if the file is not within the theme directory.
+	if ( ! isset( $relative_path ) ) {
+		return file_get_contents( $template_file_path );
+	}
+
+	// Bypass cache while developing a theme.
+	$can_use_cache = ! wp_is_development_mode( 'theme' );
+	if ( ! $can_use_cache ) {
+		return file_get_contents( $template_file_path );
+	}
```

Comment:
What about combining these for the early return? It would save on a variable, inverted logic for setting / checking `$can_use_cache`, and an extra `if` block to do the same thing.

```php
// Bypass cache if the file is not in the theme directory or a theme is being developed.
if ( ! isset( $relative_path ) || wp_is_development_mode( 'theme' ) {
	return file_get_contents( $template_file_path );
}
```

Alternatively, we can still save on the variable and the inverted logic by doing this for the second `if` block since the inline comment already provides the explanation.
```php
// Bypass cache while developing a theme.
if ( wp_is_development_mode( 'theme' ) ) {
	return file_get_contents( $template_file_path );
}
```
Not a blocker though.

Reply:
Done in https://github.com/WordPress/wordpress-develop/pull/5463/commits/a262dab4ee6cdd1d0445a8f5994bcdc939111a3b.

I didn't go with your first suggestion because in https://github.com/WordPress/wordpress-develop/pull/5463/commits/4414d8b1c5c6e4245ff47e791e820f30d1f62e44 I had to add another important consideration related to this clause, which only applies to the development mode check.